### PR TITLE
[agent-b][issue #1565] Add claude_cli doctor preflight

### LINKED ISSUE #1565: [dev][doctor] Preflight claude_cli local proof prerequisites
Part of #1441.

## Problem

Local `claude_cli` proof runs still require too much hidden setup knowledge before an operator can tell whether the machine is ready to run `swarm serve --dev` or `swarm run --backend claude_cli`.

This child covers the diagnostic/preflight surface only. It should make the current setup state explicit before any serve/run attempt.

## Scope

Define and implement a canonical preflight/doctor check for local `claude_cli` proof prerequisites.

The check should cover:

- Docker is reachable.
- The resolved workspace image exists.
- The resolved workspace image can execute the configured Claude CLI command.
- Required secrets for `claude_cli` are present through the existing secrets/credential owner.
- API/MCP ports required by the selected local flow are available or conflicts are reported clearly.
- Stale `SWARM_TOOL_GATEWAY_URL` / `SWARM_TOOL_GATEWAY_CONTAINER_URL` values are detected and reported with actionable remediation.
- Retired or unsupported backend/runtime env vars are rejected or reported according to the current source authority.

## Source authority / constraints

- Parent: #1441.
- Must consume existing `swarm secrets check` / credential store authority for secret presence. Do not invent a separate credential checker.
- Must stay aligned with `swarm serve --dev`; this should be the local proof profile owner, not a new independent profile system.
- If a new command surface is proposed, the gate must decide whether it is `swarm doctor --backend claude_cli`, a `serve --dev` preflight mode, or both.

## Non-goals

- Do not build workspace images here.
- Do not write `.swarm/connection.json` / `.swarm/local.yaml` here.
- Do not introduce `swarm dev serve` here.
- Do not silently change stale gateway env precedence here unless the stale-env child explicitly lands that policy.
- Do not change deployment-mode `swarm serve` behavior as a side effect.

## Proof requirements

Before implementation, post a Pre-Implementation Coverage Audit.

The eventual PR must prove:

- Missing Docker produces one actionable diagnostic.
- Missing workspace image produces one actionable diagnostic.
- Workspace image present but Claude CLI missing produces one actionable diagnostic.
- Missing `claude_cli` secret is reported through the existing secrets/credential owner.
- Stale MCP gateway env values are reported without requiring the operator to discover the issue through a later runtime failure.
- The supported surface works without direct DB inspection.

### PR DESCRIPTION
**Pull request by yazzaoui**
Closes #1565

## Summary

Adds a spec-backed local `claude_cli` preflight/admission owner and exposes it through `swarm doctor --backend claude_cli`. The same owner now feeds the relevant fail-closed startup path for `swarm serve --dev --backend claude_cli` and local foreground `swarm run --backend claude_cli` through the serve startup owner.

The preflight reports typed findings for backend credentials, contract-required secrets, Docker/workspace image availability, Claude CLI availability inside the workspace image, listener availability, and stale MCP gateway env values. `doctor` renders text or JSON and does not start runtime or read/write runtime DB state.

## Governing refs/context

Exact governing spec sections:

- `platform-spec.yaml` `cli_specification.foundations.local_claude_cli_preflight_admission`
- `platform-spec.yaml` `command_catalog.doctor`
- `platform-spec.yaml` `engine.agent_session_management.llm_provider_selection_config_authority`
- `platform-spec.yaml` `tool_model.credential_store`
- `platform-spec.yaml` `cli_specification.foundations.local_cli_test_gateway_startup`
- `platform-spec.yaml` `cli_specification.foundations.local_cli_test_workspace_cli_availability`
- `platform-spec.yaml` `cli_specification.foundations.api_connection_auth_config_precedence`
- `platform-spec.yaml` `command_catalog.serve`

Binding issue/gate context:

- #1565 issue body and locked design comment
- #1565 Pre-Implementation Coverage Audit
- #1565 independent gate outcome: approved as first slice
- Parent tracker: #1441

## Semantic migration

New canonical owner: `platform-spec.yaml#cli_specification.foundations.local_claude_cli_preflight_admission`, implemented by `cmd/swarm local claude_cli preflight/admission owner`.

Old producer/reader/writer paths now invalid: doctor-only prerequisite helpers, run-only prerequisite helpers, direct contract-secret lookup outside the credential store, direct backend credential interpretation outside backend profile authority, direct Docker/image/CLI probes outside the workspace lifecycle owner, and stale gateway checks that do not consume the serve startup rule.

Production paths checked for surviving old behavior: `swarm doctor`, `swarm serve --dev`, local foreground `swarm run`, non-dev disk-contract `swarm serve --backend claude_cli` runtime-safety subset, `swarm secrets check`, workspace manager prerequisites, backend selection/config loading, serve listener binding, and serve MCP gateway env validation.

Migration completeness: complete for the #1565 chosen working failure class. Parent #1441 remains open for #1566 workspace image build/validation, #1567 API connection discovery, #1568 stale gateway env policy, and documentation/local proof tails.

## Verification

- `go test ./cmd/swarm ./internal/runtime/workspace`
- `go test ./internal/store` after an initial unrelated SQLite lock timing failure in the first whole-suite attempt
- `go test ./...`

### CODE REVIEW

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit

### Semantic concepts changed

| Concept | Change |
| --- | --- |
| Local `claude_cli` prerequisite findings | Added one typed finding/report model with category, status, severity, remediation, and owner. |
| `swarm doctor --backend claude_cli` | Added supported reporting surface that renders shared findings without starting runtime or touching runtime DB state. |
| Serve/run local admission | Wired `swarm serve --dev --backend claude_cli` and local foreground `swarm run --backend claude_cli` through the same preflight owner before store selection/readiness when blockers exist. |
| Workspace capability proof | Exposed workspace manager checks for Docker availability, resolved image availability, and configured Claude CLI command availability inside the image. |
| Spec/source authority | Promoted `cli_specification.foundations.local_claude_cli_preflight_admission` and `command_catalog.doctor` in `platform-spec.yaml`. |

Chosen working failure class closed: local `claude_cli` proof prerequisites were not represented by one supported preflight/admission owner shared by `swarm doctor`, `swarm serve --dev`, and local foreground `swarm run`.

Parent failure class above it: #1441 local realistic proof setup remains too hidden/profile-fragmented across workspace image setup, local connection discovery, stale gateway policy, and docs.

Broader failure class: runtime-operations/local proof ergonomics for supported backends and workspaces.

Issue was symptom-shaped: yes. The stale/local proof failures were entry evidence; the PR boundary is the full #1565 prerequisite owner and its approved consumers.

Achieved closure level: chosen working failure class closed for #1565; parent #1441 remains open.

Parent remaining after this PR: yes. Tail estimate unchanged from gate: #1566 workspace image build/validation, #1567 API connection discovery, #1568 stale gateway env policy, plus docs/local proof cleanup. Rough grouping remains three focused PRs plus docs cleanup; confidence medium. Implementation did not reveal a new same-class sibling beyond the approved split list.

### Canonical owners after change

| Owner | Canonical role | Touched consumers |
| --- | --- | --- |
| `platform-spec.yaml#cli_specification.foundations.local_claude_cli_preflight_admission` / `cmd/swarm local claude_cli preflight/admission owner` | Top-level typed preflight/admission owner for #1565 local `claude_cli` prerequisites. | `swarm doctor`, `swarm serve --dev`, local foreground `swarm run`, non-dev disk-contract `swarm serve --backend claude_cli` runtime-safety subset, tests. |
| `engine.agent_session_management.llm_provider_selection_config_authority` / `internal/runtime/llm/selection` | Backend profile, retired selector rejection, backend credential env requirement. | `runLocalClaudeCLIPreflight`, `loadRuntimeConfigWithOptions`, serve/run config loading. |
| `tool_model.credential_store` / `internal/runtime/credentials` and `swarm secrets check` | Contract-required secret presence and resolution. | `swarm doctor --contracts`, serve/run preflight contract-secret warning path, existing `swarm secrets check`. |
| `cli_specification.foundations.local_cli_test_workspace_cli_availability` / workspace lifecycle owner | Docker/image/workspace CLI availability. | `DockerManager.CheckDockerAvailable`, `CheckWorkspaceImageAvailable`, `CheckWorkspaceCLICommandAvailable`, existing `EnsurePrereqs`. |
| `cli_specification.foundations.local_cli_test_gateway_startup` / serve startup owner | MCP gateway env/token consistency and stale env validation. | doctor/preflight gateway finding, serve/run admission, existing `configureServeMCPGatewayEnv`. |
| Serve listener owner in `runServeRuntime` | Listener address validation and bind proof. | doctor/preflight listener findings and serve startup. |

### Systematic consumption audit

| Seam | Classification | Proof |
| --- | --- | --- |
| `swarm doctor --backend claude_cli` | moved to canonical owner in this work | `cmd/swarm/doctor.go` calls `runLocalClaudeCLIPreflight`; tests cover text/JSON. |
| `swarm serve --dev --backend claude_cli` | moved to canonical owner in this work | `runServeRuntime` calls `runServeLocalClaudeCLIPreflight` before store selection; test proves blocker stops before DB/store. |
| Local foreground `swarm run --backend claude_cli` | moved to canonical owner in this work | `startLocalRunServe` sets `LocalRun`; existing run path consumes serve startup; test asserts marker. |
| Non-dev disk-contract `swarm serve --backend claude_cli` | moved to runtime-safety subset in this work | `shouldRunServeLocalClaudeCLIPreflight` runs for non-bundle disk-contract serve; non-`claude_cli` backends skip inside owner. |
| Persisted `--bundle-hash` serve | different/split behavior | explicitly skipped because it is not local disk-contract proof setup; existing runtime bundle checks remain authoritative. |
| Backend credential/env selector checks | already consumes canonical owner | `loadRuntimeConfigWithOptions` rejects retired env selectors; preflight uses `LLMBackendProfile` and `llmselection.RequireCredential`. |
| Contract secret checks | already consumes canonical owner | preflight delegates to `buildCredentialStore` and `runtimecredentials.MissingRequired`, same requirement owner as `swarm secrets check`. |
| Workspace Docker/image checks | already consumes canonical owner, exposed for shared preflight | `EnsurePrereqs` now delegates to exported check methods; preflight calls those methods. |
| Workspace Claude CLI capability | moved to canonical owner in this work | new `DockerManager.CheckWorkspaceCLICommandAvailable` executes `command -v` inside the resolved image. |
| Gateway stale env validation | already consumes canonical owner | preflight calls existing `validateExistingServeGatewayURL`; no precedence/auto-fix change. |
| Listener availability | already consumes serve listener owner | preflight uses existing `listenServeHTTPListener` and closes the listener. |
| Host workspace `claude_cli` | split / separate class | reported as unsupported; #1138/#1213 remain the host execution parity track. |
| Image build/pull automation | split / separate class | #1566 remains open; this PR only detects/report missing or unusable image. |
| API connection discovery files | split / separate class | #1567 remains open; this PR writes no `.swarm/connection.*`. |
| Stale gateway env policy/auto-fix | split / separate class | #1568 remains open; this PR detects/reports/fails closed under current serve rule only. |

Old paths now invalid/non-authoritative: doctor-only duplicate checkers, run-only duplicate checkers, direct contract-secret readers outside credential-store/MissingRequired, direct backend credential inference outside backend profile authority, direct Docker/image/CLI probes outside workspace lifecycle methods, and gateway stale-env checks that do not consume the serve startup validator.

Old paths still surviving as authoritative subordinate owners: `swarm secrets check`, backend selection/config loading, workspace lifecycle `EnsurePrereqs`, runtime Claude startup validation, serve listener binding, and serve MCP gateway env derivation/validation.

### Manifestation coverage table

| Manifestation | Status | Exact proof |
| --- | --- | --- |
| No supported preflight command diagnoses local `claude_cli` readiness. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsMissingPrerequisites`; `TestDoctorClaudeCLIPreflightJSONReportsOKWithoutDB`. |
| Missing Docker only surfaces later through workspace/startup failure. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsMissingDocker`. |
| Missing resolved workspace image only surfaces through startup failure. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsMissingPrerequisites` with `SWARM_TEST_DOCKER_IMAGE_MISSING=1`. |
| Image exists but configured Claude CLI command is unavailable. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsMissingWorkspaceCLI`. |
| Missing `CLAUDE_CODE_OAUTH_TOKEN` backend credential is not preflighted. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsMissingPrerequisites`; `TestRunServeRuntimeConsumesLocalClaudePreflightBeforeStoreSelection`. |
| Contract-required secret presence must use existing credential-store authority. | reproduced and fixed | `TestDoctorClaudeCLIPreflightUsesCredentialStoreForContractSecrets`. |
| Stale `SWARM_TOOL_GATEWAY_URL` / `SWARM_TOOL_GATEWAY_CONTAINER_URL` discovered late. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsStaleGatewayEnv`. |
| Retired backend/runtime env selectors must fail closed under backend-selection authority. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsRetiredBackendEnv`. |
| Doctor/reporting path must not require runtime DB state. | reproduced and fixed | `TestDoctorClaudeCLIPreflightJSONReportsOKWithoutDB` sets a must-not-be-used SQLite path and completes through local checks only. |
| Serve admission must stop before store selection on local blocker. | reproduced and fixed | `TestRunServeRuntimeConsumesLocalClaudePreflightBeforeStoreSelection` uses invalid store mode and proves preflight fails first. |
| Local foreground run must consume the serve startup owner instead of forking a run-only checker. | execution-proven through the same corrected path | `TestRunCommandLocalForegroundConsumesServeOwnerAndV1API` asserts `serveOpts.LocalRun`. |
| Non-`claude_cli` backends should not inherit `claude_cli` proof prerequisites. | execution-proven through the same corrected path | `runLocalClaudeCLIPreflight` returns `backend_not_claude_cli` skipped before loading contracts/workspace. Covered by code path and focused package tests. |
| Persisted bundle serve, image build, connection discovery, stale-env policy rewrite, host workspace `claude_cli`. | split / escalated as separate class | Existing split trackers #1566, #1567, #1568, #1138/#1213; PR code keeps these paths out of scope. |

### Sibling contexts checked

Checked #1441 parent, #1566 image build/validation, #1567 API connection discovery, #1568 stale gateway env policy, #1138/#1213 host workspace execution, backend selection #1128/#1129 closure, existing secrets command owner, workspace lifecycle owner, serve gateway owner, and local foreground run serve startup path.

Generic failing proof used/created: created typed CLI tests with fake Docker behavior for missing Docker, missing image, missing Claude CLI command, stale gateway env, missing backend credential, missing contract secret, retired env selector, JSON/no-DB output, and serve-before-store admission.

Watchlist decision: no repository watchlist file update in this PR. #1565 remains mapped under runtime-operations/local `claude_cli` proof prerequisites via #1441 and sibling issues. The concrete parent/sibling tracking is in #1441/#1566/#1567/#1568, and no new sibling was discovered during implementation.

### Architecture feedback

Deeper smell: prerequisite facts were previously scattered as startup side effects instead of one typed prerequisite model with category, owner, severity, surface, and remediation. This PR creates the first shared owner for `claude_cli`, but the broader diagnostic architecture is still command-local rather than a general all-backend readiness framework.

Long-run direction: keep subordinate source owners intact, but promote reusable prerequisite checks as typed findings with per-consumer severity projection. `doctor` should render/report; `serve` and local `run` should fail closed on admission blockers.

Tracking decision: concrete #1565 slice is tracked and closed by this PR. Broader diagnostic expansion remains on #1441 and sibling issues unless a new multi-backend doctor/diagnostic issue is later justified by evidence.

Effort/ROI estimate for architecture smell: medium effort, high ROI if expanded deliberately after #1441 tails. A reusable check registry across backends would likely be 2-4 focused PRs: owner extraction, backend-specific adapters, command rendering/API shape if promoted, and cross-surface tests. ROI is high because it prevents doctor/serve/run drift, but doing it now would exceed the approved first-slice boundary.

### Required supported-surface proof actually run

- `go test ./cmd/swarm ./internal/runtime/workspace` passed.
- First `go test ./...` attempt hit an unrelated timing-sensitive SQLite lock test in `internal/store`; no touched files were in that package.
- `go test ./internal/store` passed on isolated rerun.
- Final `go test ./...` passed.

File: cmd/swarm/local_preflight.go [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+		report.add(localPreflightBackendPrerequisite, "backend_profile_invalid", localPreflightSeverityBlocker, localPreflightStatusFailed, err.Error(), "set a supported backend with --backend or llm.backend")
+		return report.finalize()
+	}
+	report.Backend = strings.TrimSpace(profile.ID)
+	if profile.ID != llmselection.BackendClaudeCLI {
+		report.add(localPreflightBackendPrerequisite, "backend_not_claude_cli", localPreflightSeverityInfo, localPreflightStatusSkipped, fmt.Sprintf("backend %q does not require claude_cli local proof prerequisites", profile.ID), "")
+		return report.finalize()
+	}
+
+	if err := llmselection.RequireCredential(profile, os.LookupEnv); err != nil {
```
[the remark this answers is not in the document]
Comment (yazzaoui) [reply cmd/swarm/local_preflight.go]: Addressed in d4abaded.

The preflight now loads the selected source and contract-secret requirements before applying the backend credential gate. If the source declares no agents, it records `backend_credential_skipped_agent_free` and `agent_free_source` as skipped findings and returns successfully without requiring `CLAUDE_CODE_OAUTH_TOKEN` or probing workspace Claude CLI availability. Agent-present sources still use `llmselection.RequireCredential` and the workspace checks.

I also updated `platform-spec.yaml` so the promoted preflight owner matches runtime’s declared-agent credential gate.

Proof run:
- `go test ./cmd/swarm ./internal/runtime/workspace`
- `go test ./...`

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit Addendum

Review feedback addressed: agent-free selected sources must not require `CLAUDE_CODE_OAUTH_TOKEN` for `swarm doctor --backend claude_cli` or the shared preflight owner.

Semantic correction: backend selector authority is still consumed for `claude_cli`, but backend credential authority is now applied only after the selected source is loaded and only when `source.AgentEntries()` is non-empty. This matches the existing runtime gate in `validateSelectedBackendCredentialForDeclaredAgents`.

Owner update: `platform-spec.yaml#cli_specification.foundations.local_claude_cli_preflight_admission` now explicitly records the declared-agent credential rule. Contract-required secrets still consume the credential store independently and are checked before the agent-free backend/workspace skip.

Manifestation table addendum:

| Manifestation | Status | Exact proof |
| --- | --- | --- |
| Agent-free `claude_cli` doctor/preflight wrongly blocks on missing backend credential. | reproduced and fixed | `TestDoctorClaudeCLIPreflightSkipsCredentialForAgentFreeContracts` proves exit 0 with empty `CLAUDE_CODE_OAUTH_TOKEN`, skipped backend credential finding, skipped workspace proof, and no `missing_backend_credential`. |
| Agent-present `claude_cli` still requires backend credential. | execution-proven through same corrected path | Existing `TestDoctorClaudeCLIPreflightReportsMissingPrerequisites` and `TestRunServeRuntimeConsumesLocalClaudePreflightBeforeStoreSelection` still pass. |

Proof run after review fix:

- `go test ./cmd/swarm ./internal/runtime/workspace`
- `go test ./...`

File: internal/runtime/workspace/manager.go [hunk 2]
```diff
@@ -266,18 +266,50 @@ func (m *DockerManager) EnsurePrereqs(ctx context.Context) error {
...
+	}
+	image := strings.TrimSpace(m.cfg.WorkspaceImage)
+	if image == "" {
+		return fmt.Errorf("workspace prerequisite failed: workspace image is required")
+	}
+	command = strings.TrimSpace(command)
+	if command == "" {
+		command = "claude"
+	}
+	if _, err := m.RunDocker(ctx, "run", "--rm", "--entrypoint", "sh", image, "-lc", `command -v -- "$1" >/dev/null`, "swarm-cli-proof", command); err != nil {
```
Comment (yazzaoui) [inline internal/runtime/workspace/manager.go]: This only proves that the configured name resolves in the image; it does not prove the workspace image can execute the configured Claude CLI command. The governing spec says the workspace CLI availability owner must prove the resolved workspace can execute the configured command before readiness, and #1565’s manifestation is “image present but configured Claude CLI command unavailable.” A binary can be present in PATH but fail on startup due to a bad interpreter, missing dynamic loader, broken installation, or other launch failure, so `doctor` / serve admission can pass while the real Claude startup path still fails later.

Please change this to an actual execution proof, e.g. `command -v "$1" && "$1" --version` or another explicitly approved harmless execution probe, and add a test where `command -v` would succeed but invoking the configured command exits non-zero.

**Review by yazzaoui — COMMENTED** [review]
Process review result: changes needed. GitHub would not allow `REQUEST_CHANGES` from this authenticated account because it is the PR author, but this is a blocking review finding under the repo process.

I verified the issue record, pre-audit, independent gate, PR proof audit/addendum, current head, unresolved review thread state, cited spec sections, and split tracker state. The issue category is correct and the chosen #1565 class is acceptable as a first slice of #1441: a shared local `claude_cli` preflight/admission owner consumed by `doctor`, `serve --dev`, local foreground `run`, and the disk-contract runtime-safety subset. Parent/sibling trackers #1441, #1566, #1567, #1568, #1138, and #1213 are open and independently verified. The prior unresolved agent-free inline thread is obsolete on current head and I resolved it: the source is loaded before backend credential/workspace checks and agent-free sources skip both.

Blocking gap: the workspace CLI capability manifestation is not actually proven. `platform-spec.yaml#cli_specification.foundations.local_cli_test_workspace_cli_availability` requires proving the resolved workspace can execute the configured Claude CLI command; #1565 also asks for the resolved workspace image to execute the configured command. Current head only runs `command -v` inside the image. That proves lookup, not executable startup. A broken executable, bad shebang/interpreter, missing dynamic loader, or unusable CLI binary can still pass the preflight and leave `swarm doctor` / serve admission claiming the `workspace_claude_cli_available` prerequisite is OK until the later Claude startup path fails.

Required before merge: make the preflight capability proof execute the configured command through an approved harmless probe, for example the parent-tracker expected equivalent of `command -v <cmd> && <cmd> --version`, or another spec-backed execution probe if `--version` is not the chosen contract. Add a regression test where lookup succeeds but executing the configured command fails, and update the PR proof audit so the manifestation row proves execution rather than path lookup.

**Comment by yazzaoui** [comment]
## Process checklist review

- Guideline checks: issue #1565, locked design, pre-audit, gate, PR proof audit/addendum, current head, cited spec sections, and inline thread state reviewed.
- Prior unresolved thread: agent-free credential/workspace finding verified obsolete on `d4abaded` and resolved.
- Spec compliance: incomplete; workspace CLI proof uses `command -v` lookup rather than executing the configured Claude CLI command required by the cited workspace CLI availability owner.
- Failure-class coverage: incomplete for the `workspace_claude_cli_available` manifestation; the chosen class remains acceptable, but proof is under-strength.
- Residual risk: high for false-positive doctor/serve admission on broken CLI installations inside an otherwise present image.
- Closure level: not merge-ready; local preflight owner is mostly wired, but the chosen failure class is not proven eliminated.
- Merge recommendation: do not merge until the preflight executes an approved harmless Claude CLI command probe, adds lookup-succeeds/execution-fails regression coverage, and updates the proof audit.

File: internal/runtime/workspace/manager.go (hunk 2 above)
Comment (yazzaoui) [reply internal/runtime/workspace/manager.go]: Addressed in e26737ae.

`CheckWorkspaceCLICommandAvailable` now proves both lookup and execution by running `command -v -- "$1" >/dev/null && "$1" --version >/dev/null` inside the resolved workspace image. The diagnostic now says the configured Claude CLI command cannot execute, rather than only being unavailable.

Added `TestDoctorClaudeCLIPreflightReportsBrokenWorkspaceCLI`, where the Docker stub simulates a command that lookup would resolve but `--version` exits non-zero. That test would not fail against the old command-v-only probe, and now proves the stronger execution check.

Proof run:
- `go test ./cmd/swarm ./internal/runtime/workspace`
- `go test ./...`

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit Addendum

Review feedback addressed: workspace CLI capability proof must execute the configured Claude CLI command, not only prove the name resolves in `PATH`.

Semantic correction: `internal/runtime/workspace.DockerManager.CheckWorkspaceCLICommandAvailable` now runs a harmless execution probe inside the resolved workspace image: `command -v -- "$1" >/dev/null && "$1" --version >/dev/null`. This keeps the workspace lifecycle owner as the canonical subordinate owner while strengthening the #1565 local preflight/admission finding from name lookup to actual command executability.

Manifestation table addendum:

| Manifestation | Status | Exact proof |
| --- | --- | --- |
| Workspace image contains a resolvable `claude` name but the configured command cannot execute. | reproduced and fixed | `TestDoctorClaudeCLIPreflightReportsBrokenWorkspaceCLI` simulates command lookup success with `--version` execution failure and verifies `workspace_claude_cli_unavailable` plus actionable remediation. |
| Missing configured Claude CLI command remains reported. | execution-proven through same corrected path | `TestDoctorClaudeCLIPreflightReportsMissingWorkspaceCLI` still passes. |
| Valid workspace image with runnable configured Claude CLI remains accepted. | execution-proven through same corrected path | `TestDoctorClaudeCLIPreflightJSONReportsOKWithoutDB` still passes. |

Proof run after review fix:

- `go test ./cmd/swarm ./internal/runtime/workspace`
- `go test ./...`

**Review by yazzaoui — COMMENTED** [review]
Process review result: no findings; merge recommendation approve.

I re-reviewed current head e26737ae against #1565, the locked design/pre-audit/gate, the post-implementation proof audit/addenda, the unresolved review threads, and the cited spec sections. The previous blocking workspace CLI proof gap is fixed: `DockerManager.CheckWorkspaceCLICommandAvailable` now proves lookup plus execution with `command -v -- "$1" >/dev/null && "$1" --version >/dev/null`, and `TestDoctorClaudeCLIPreflightReportsBrokenWorkspaceCLI` covers lookup success with execution failure. I resolved the old workspace CLI review thread as obsolete on current head.

Issue category remains correct: source-authority/supported-surface parity gap for local `claude_cli` readiness diagnostics. The chosen #1565 failure class is acceptable as a first slice of #1441, and the shared owner is consumed by `doctor`, `serve --dev`, local foreground `run`, and the disk-contract runtime-safety subset. Parent/sibling trackers #1441, #1566, #1567, #1568, #1138, and #1213 remain open and verified.

Spec compliance: satisfied for the #1565 scope. Failure-class coverage: satisfied for known manifestations after the addenda. Concept-wide drift-reduction: material; no hidden same-concept interpreter found. Closure level: chosen failure class eliminated; parent #1441 remains open for tracked siblings. No new follow-up issue or watchlist change is needed.

**Comment by yazzaoui** [comment]
## Process checklist review

- Guideline checks: issue #1565, locked design, pre-audit/gate, PR proof audit/addenda, cited spec sections, current head `e26737ae`, and review thread state reviewed.
- Prior blocker: workspace CLI probe now proves lookup plus execution with `"$1" --version`; regression covers lookup-success/execution-failure.
- Spec compliance: pass for #1565 scope.
- Failure-class coverage: pass for known manifestations.
- Residual risk: parent #1441 tails remain tracked by #1566/#1567/#1568/#1138/#1213; no new same-class residual found.
- Risk level: medium because local runtime prereq/admission is operator-facing, but proof is now adequate.
- Closure level: #1565 chosen failure class eliminated; parent #1441 remains open.
- Merge recommendation: approve / merge-ready, subject to normal CI.
